src/apps/spoc_verifier.py

```diff
@@ -11,44 +12,41 @@
 
 main_col, geo_col, action_col = st.beta_columns([2, 1, 0.5])
 
-species = pd.DataFrame(
-    [
-        {
-            "Paper ID": "A",
-            "Instance ID": "A02",
-            "Species - Genus": None,
-            "Time": None,
-            "Place": None,
-        },
-        {
-            "Paper ID": "A",
-            "Instance ID": "A03",
-            "Species - Genus": "Tonicella lineata",
-            "Time": "1982",
-            "Place": "Grimes Point",
-        },
-        {
-            "Paper ID": "B",
-            "Instance ID": "B01",
-            "Species - Genus": None,
-            "Time": None,
-            "Place": None,
-        },
-    ]
-)
+species = pd.read_json("data/species-records.json")
+
+grid_options = GridOptionsBuilder.from_dataframe(species)
+
+# Grid options
+grid_options.configure_selection("single", use_checkbox=True)
+grid_options.configure_pagination(paginationAutoPageSize=True)
 
 with main_col:
     """
     # SPOC Verifier
     """
-    st.dataframe(species)
+    records_grid = AgGrid(
+        species,
+        gridOptions=grid_options.build(),
+        data_return_mode=DataReturnMode.FILTERED,
+        update_mode=GridUpdateMode.MODEL_CHANGED,
+        allow_unsafe_jscode=True,
+    )
+
+selected = records_grid["selected_rows"]
+selected_df = pd.DataFrame(selected)
 
 with geo_col:
-    """
-    # A-A03
-    [PLACE]
+    if not selected_df.empty:
+        f"""
+        ## {selected_df['Species'][0]}
 
-    """
+        ## Places
+        """
+        for place in selected_df["Place"]:
+            place = eval(place)
```

Comment:
This `exec` call should be temporary, when users are able to edit the AgGrid, we'll want to sanitize `place`.